cmake/modules/AddSwift.cmake

```diff
@@ -1055,8 +1052,17 @@ function(_add_swift_library_single target name)
   set(link_flags ${SWIFTLIB_SINGLE_LINK_FLAGS})
   set(library_search_directories
       "${SWIFTLIB_DIR}/${SWIFTLIB_SINGLE_SUBDIR}"
-      "${SWIFT_NATIVE_SWIFT_TOOLS_PATH}/../lib/swift/${SWIFTLIB_SINGLE_SUBDIR}"
+      "${SWIFT_NATIVE_SWIFT_TOOLS_PATH}/../lib/swift/${SWIFTLIB_SINGLE_SUBDIR}")
+
+  if(IS_DARWIN)
+    # Since we lipo each arch together on Darwin, search in the common directory
+    list(APPEND library_search_directories
       "${SWIFT_NATIVE_SWIFT_TOOLS_PATH}/../lib/swift/${SWIFT_SDK_${SWIFTLIB_SINGLE_SDK}_LIB_SUBDIR}")
+  else()
+    # No fat binaries on other archs so search in an arch specific directory
```

Comment:
s/archs/targets/ ?

Reply:
good catch. English is my first language but I'm still terrible at it. Probably why I stuck to math in school.